components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -208,16 +214,11 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
                     Claim claimObj = claimManager.getClaim(missingClaim.getValue());
                     if (claimObj != null && claimObj.isReadOnly()) {
                         if (LoggerUtils.isDiagnosticLogsEnabled()) {
-                            Map<String, Object> params = new HashMap<>();
-                            params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER,
-                                    context.getServiceProviderName());
-                            params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, context.getTenantDomain());
-                            params.put(FrameworkConstants.LogConstants.MISSING_CLAIMS, missingClaim);
-                            LoggerUtils.triggerDiagnosticLogEvent(
-                                    FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params,
-                                    LogConstants.FAILED, "One or more read-only claim is missing in the requested " +
-                                            "claim set",
-                                    FrameworkConstants.LogConstants.ActionIDs.HANDLE_MISSING_CLAIMS, null);
+                            diagnosticLogBuilder.inputParam("read only claim", claimObj.getClaimUri())
```

Comment:
shall we use a constant for this `read only claim`

Reply:
Fixed with d71d762bcbd0f1c2101e584037685579249682b0